openwisp_controller/subnet_division/tests/test_admin.py

```diff
@@ -212,38 +218,24 @@ def test_subnet_filter_mutitenancy(self):
             organization=org2,
             master_subnet=master_subnet2,
         )
-        config2 = self._create_config(
+        self.config2 = self._create_config(
             device=self._create_device(name='org2-device', organization=org2)
         )
-        config2.templates.add(template2)
-
-        org2_owner = self._create_org_user(
-            organization=org2, is_admin=True, user=self._create_operator()
-        )
-        org2_owner.user.user_permissions.add(*self.get_operator_permissions())
+        self.config2.templates.add(template2)
+        administrator = self._create_administrator([org2])
         self.client.logout()
-        self.client.force_login(org2_owner.user)
+        self.client.force_login(administrator)
 
+    def test_device_filter_mutitenancy(self):
```

Comment:
This test is failing. I checked the response and it is returning only one subnet with organization org2 but the device name is not present in that:

![image](https://user-images.githubusercontent.com/56037184/157085340-21e11c88-fe3d-4908-924b-30eae1b7cf11.png)

Earlier it was showing because the operator had all the permissions.

Reply:
This class should inherit `TransactionTestCase` because subnet division is wrapped in atomic transactions.

Reply:
But then Group queryset will be empty.

Reply:
We should add fixtures for this test.